hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/StreamSync.java

```diff
@@ -457,7 +457,7 @@ public Pair<Option<String>, JavaRDD<WriteStatus>> syncOnce() throws IOException
 
   private Option<String> getLastPendingClusteringInstant(Option<HoodieTimeline> commitTimelineOpt) {
     if (commitTimelineOpt.isPresent()) {
-      Option<HoodieInstant> pendingClusteringInstant = commitTimelineOpt.get().filterPendingReplaceTimeline().lastInstant();
+      Option<HoodieInstant> pendingClusteringInstant = commitTimelineOpt.get().getLastPendingClusterCommit();
```

Comment:
Should we revisit all usage of `filterPendingReplaceTimeline` in a follow-up and see if some should be fixed by differentiating clustering vs non-clustering replace commits?

Reply:
The usage in BaseHoodieTableServiceClient looks bad. I wonder if we should think about changing the name in the timeline so we don't need to deserialize the instants a bunch of times

Reply:
Adding new instant types to differentiate clustering vs non-clustering replacecommits would make it more performant, but it also incurs format changes on storage.  For now, let's fix the correctness first.